src/plugins/vis_builder/public/visualizations/table/components/table_viz_options.tsx

```diff
@@ -0,0 +1,227 @@
+      defaultMessage: 'Max',
+    }),
+  },
+  {
+    value: AggTypes.COUNT,
+    text: i18n.translate('visTypeTableNewNew.totalAggregations.countText', {
+      defaultMessage: 'Count',
+    }),
+  },
+];
+
+function TableVizOptions() {
+  const styleState = useTypedSelector((state) => state.style) as TableOptionsDefaults;
+  const { aggConfigs } = useAggs();
+  const dispatch = useTypedDispatch();
+
+  const setOption = useCallback(
+    (callback: (draft: Draft<typeof styleState>) => void) => {
+      const newState = produce(styleState, callback);
+      dispatch(setStyleState<TableOptionsDefaults>(newState));
+    },
+    [dispatch, styleState]
+  );
+
+  const percentageColumns = useMemo(() => {
+    const defaultPercentageColText = {
+      value: '',
+      text: i18n.translate('visTypeTableNew.params.defaultPercentageCol', {
+        defaultMessage: 'Don’t show',
+      }),
+    };
+    return aggConfigs
+      ? [
+          defaultPercentageColText,
+          ...tabifyGetColumns(aggConfigs.getResponseAggs(), true)
+            .filter((col) => get(col.aggConfig.toSerializedFieldFormat(), 'id') === 'number')
+            .map(({ name }) => ({ value: name, text: name })),
+        ]
+      : [defaultPercentageColText];
+  }, [aggConfigs]);
+
+  useEffect(() => {
+    if (
+      !percentageColumns.find(({ value }) => value === styleState.percentageCol) &&
+      percentageColumns[0] &&
+      percentageColumns[0].value !== styleState.percentageCol
+    ) {
+      setOption((draft) => {
+        draft.percentageCol = percentageColumns[0].value;
+      });
+    }
+  }, [percentageColumns, styleState.percentageCol, setOption]);
+
+  const isPerPageValid = styleState.perPage === '' || styleState.perPage > 0;
+
+  return (
+    <>
+      <Option
+        title={i18n.translate('visTypeTableNewNew.params.settingsTitle', {
+          defaultMessage: 'Settings',
+        })}
+        initialIsOpen
+      >
+        <NumberInputOption
+          label={
+            <>
+              {i18n.translate('visTypeTableNewNew.params.perPageLabel', {
+                defaultMessage: 'Max rows per page',
+              })}
+              ,
+              <EuiIconTip
+                content={
+                  <FormattedMessage
+                    id="visTypeTableNewNews.field.emptyTooltip"
+                    defaultMessage="Leaving this field empty means it will use number of buckets from the response."
+                  />
+                }
+                position="right"
+              />
+            </>
+          }
+          isInvalid={!isPerPageValid}
+          min={1}
+          paramName="perPage"
+          value={styleState.perPage}
+          setValue={(_, value) =>
+            setOption((draft) => {
+              draft.perPage = value;
+            })
+          }
+        />
+
+        <SwitchOption
+          label={i18n.translate('visTypeTableNewNew.params.showMetricsLabel', {
```

Comment:
this could use a tooltip, because I don't really understand what it does.

Reply:
yeah we could add a tooltip to explain these options. Currently, it is just set exactly the same as old table vis. 
I think we could either add in the README or add a tooltip. I will add a follow up issue here.

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2741